UKHomeOffice/cop-ui
# COP-9543: Identify undefined current group in local storage

### Description
The current group can be set to "undefined" in `localStorage` but no account was being made of this - it was always expected to be a valid JSON object - and so was failing when attempting to parse it.

This was appearing when the user _has_ no current group because they have an end-dated team and they attempt to click on the "My profile" link when they're already on the profile edit page.

https://support.cop.homeoffice.gov.uk/browse/COP-9543

### To test
As outlined in https://support.cop.homeoffice.gov.uk/browse/COP-9543, starting with an end-dated team. With this fix in place, there should be no white screen of death.
